### DISCUSSION TITLE
Unhandled rendering route - Cancel rendering route

### DISCUSSION THREAD

**Discussion by funk10101** [vercel/next.js#43445] [Help]
I've been getting this intermittently. I'm working with SuperTokens about this as it happens while using the SignIn/SignUp function. But I thought I'd post here in case someone has come across this somewhere.

Basically, I click the SignIn/SignUp button the page displays.
I click the "SignUp" link, and the page displays.
I click the SignIn/SignUp button again and then I get that error.
 Please see screen shots.
![SignInScreen](https://user-images.githubusercontent.com/119277851/204218991-793e15a5-d10e-428a-a633-e6b2a58e8f66.jpg)
![SignUpScreen](https://user-images.githubusercontent.com/119277851/204219009-589e09e2-03f4-4aac-885c-ab81f163f1f1.jpg)
![SignUpScreen-Error](https://user-images.githubusercontent.com/119277851/204219025-ff291782-a82e-4f00-aa30-48a35d8429fe.jpg)

**Comment by icyJoseph** [comment]
Have you got a repository to look at? Do you observe this with React Strict mode turned off?

In general this error happens when the router "gives up" on a transition, to start a new one. You can capture it through a `Router.events.on('routeChangeError', (err, url) => console.log('cancelled?', err.cancelled)`, if cancelled is not `true` then we might have another issue going on, but most likely this it it.

One reason this happens is `await` on `router.push`, or some kind of memory leak, but most likely a transition being cancelled. Of course it might also be a bug, but a reproduction repository would be needed to investigate that.

**Reply by funk10101** [reply]
I have strictmode on. I'll check it turned off. Here's repo with the code in question:
https://github.com/funk10101/nextjs-supertokens-issue

**Reply by icyJoseph** [reply]
That repo's not enough though, it is missing the pages and other stuff right?

**Reply by funk10101** [reply]
Yes, I just posted what I think is the problem code regarding supertokens. As the rest of the site is nothing but "hello world" placeholders
UPDAte: I "commented out" this logic "frontendConfig.js" and I don't get this same error anymore:
` windowHandler: (oI) => {
      return {
        ...oI,
        location: {
          ...oI.location,
          setHref: (href) => {
            Router.push(href);
          },
        },
      };
    },`

**Reply by icyJoseph** [reply]
Aha, but could you please complete the reproduction demo, it's important to not make any assumptions.

**Reply by icyJoseph** [reply]
And why/how do you use this `windowHandler`? I am not familiar with this framework, but it looks like `setHref` is masking for `Router.push` which is asynchronous, yet it is "voided", successive calls to setHref would cause two push actions to batch, and one of them would be cancelled, ideally the first one.
